### ISSUE TITLE
Experimental ASM in 1.4.0 does not work with my Radeon 280x

### ISSUE THREAD

**Issue by blenessy** [Optiminer/OptiminerZcash#27]
Jan 17 22:33:35 zminer2 miner.sh[5627]: [2017-01-17 22:33:35.356] [info] Optiminer/Zcash 1.4.0 (C) Optiminer 2017
Jan 17 22:33:35 zminer2 miner.sh[5627]: [2017-01-17 22:33:35.356] [info] WatchDog: Executing "/usr/local/bin/force_reboot" if any GPU is stale for more than 180s.
Jan 17 22:33:35 zminer2 miner.sh[5627]: [2017-01-17 22:33:35.356] [info] Connecting to zcl.coinmine.pl:7007.
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.346] [info] Using experimental highly optimized kernel code for Tahiti devices.
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.346] [info] Autodetected '--intensity 5' for device 0.
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.352] [info] [GPU0] Device info: {"id": "0/0" "name": "Tahiti" "platform": "AMD" "driver": "1729.3 (VM)"}
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.363] [info] Extranonce is '8de6e105'.
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.379] [info] Mining target is 003ffffffffffffffffffffffffffffffffffffffffffffffffffffffff5d7c0
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.380] [info] Got new work.
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.433] [info] Connected to zcl.coinmine.pl:7007.
Jan 17 22:33:36 zminer2 miner.sh[5627]: [2017-01-17 22:33:36.433] [info] Got new work.
Jan 17 22:33:37 zminer2 miner.sh[5627]: [2017-01-17 22:33:37.039] [info] Using experimental highly optimized kernel code for Tahiti devices.
Jan 17 22:33:37 zminer2 miner.sh[5627]: [2017-01-17 22:33:37.039] [info] Autodetected '--intensity 5' for device 1.
Jan 17 22:33:37 zminer2 miner.sh[5627]: [2017-01-17 22:33:37.043] [info] [GPU1] Device info: {"id": "0/1" "name": "Tahiti" "platform": "AMD" "driver": "1729.3 (VM)"}
Jan 17 22:33:37 zminer2 miner.sh[5627]: [2017-01-17 22:33:37.627] [info] Using experimental highly optimized kernel code for Tahiti devices.
Jan 17 22:33:37 zminer2 miner.sh[5627]: [2017-01-17 22:33:37.627] [info] Autodetected '--intensity 5' for device 2.
Jan 17 22:33:37 zminer2 miner.sh[5627]: [2017-01-17 22:33:37.632] [info] [GPU2] Device info: {"id": "0/2" "name": "Tahiti" "platform": "AMD" "driver": "1729.3 (VM)"}
Jan 17 22:33:38 zminer2 miner.sh[5627]: [2017-01-17 22:33:38.221] [info] Using experimental highly optimized kernel code for Tahiti devices.
Jan 17 22:33:38 zminer2 miner.sh[5627]: [2017-01-17 22:33:38.221] [info] Autodetected '--intensity 5' for device 3.
Jan 17 22:33:38 zminer2 miner.sh[5627]: [2017-01-17 22:33:38.226] [info] [GPU3] Device info: {"id": "0/3" "name": "Tahiti" "platform": "AMD" "driver": "1729.3 (VM)"}
Jan 17 22:33:38 zminer2 miner.sh[5627]: [2017-01-17 22:33:38.817] [info] Using experimental highly optimized kernel code for Tahiti devices.
Jan 17 22:33:38 zminer2 miner.sh[5627]: [2017-01-17 22:33:38.817] [info] Autodetected '--intensity 5' for device 4.
Jan 17 22:33:38 zminer2 miner.sh[5627]: [2017-01-17 22:33:38.824] [info] [GPU4] Device info: {"id": "0/4" "name": "Tahiti" "platform": "AMD" "driver": "1729.3 (VM)"}
Jan 17 22:33:42 zminer2 miner.sh[5627]: [2017-01-17 22:33:42.166] [info] [GPU0]  185.0 I/s 0.0 S/s (5s)
Jan 17 22:33:42 zminer2 miner.sh[5627]: [2017-01-17 22:33:42.753] [info] [GPU1]  183.0 I/s 0.0 S/s (5s)
Jan 17 22:33:43 zminer2 miner.sh[5627]: [2017-01-17 22:33:43.333] [info] [GPU2]  182.0 I/s 0.0 S/s (5s)
Jan 17 22:33:43 zminer2 miner.sh[5627]: [2017-01-17 22:33:43.932] [info] [GPU3]  177.0 I/s 0.0 S/s (5s)
Jan 17 22:33:44 zminer2 miner.sh[5627]: [2017-01-17 22:33:44.542] [info] [GPU4]  183.0 I/s 0.0 S/s (5s)
Jan 17 22:33:45 zminer2 miner.sh[5627]: [2017-01-17 22:33:45.358] [info] [Total]  908.0 I/s 0.0 S/s (5s)

**Comment by mdevalery** [comment]
I have the same problem with r9 390

**Comment by rbiegel** [comment]
Here too on R9 280X:

```
[2017-01-18 11:37:23.937] [info] Optiminer/Zcash 1.4.0 (C) Optiminer 2017
[2017-01-18 11:37:23.938] [info] Connecting to xxxxx.
[2017-01-18 11:37:25.299] [info] Using experimental highly optimized kernel code for Tahiti devices.
[2017-01-18 11:37:25.299] [info] Autodetected '--intensity 5' for device 0.
[2017-01-18 11:37:25.317] [info] [GPU0] Device info: {"id": "0/0" "name": "Tahiti" "platform": "AMD" "driver": "1912.5 (VM)"}
[2017-01-18 11:37:25.404] [info] TLS handshake with xxxxx succeeded.
[2017-01-18 11:37:25.443] [info] Extranonce is 'xxxxx'.
[2017-01-18 11:37:25.480] [info] Connected to xxxxx.
[2017-01-18 11:37:25.482] [info] Mining target is xxxxx
[2017-01-18 11:37:26.073] [info] Using experimental highly optimized kernel code for Tahiti devices.
[2017-01-18 11:37:26.073] [info] Autodetected '--intensity 5' for device 1.
[2017-01-18 11:37:26.074] [info] GPU0 Waiting for work from pool.
[2017-01-18 11:37:26.092] [info] [GPU1] Device info: {"id": "0/1" "name": "Tahiti" "platform": "AMD" "driver": "1912.5 (VM)"}
[2017-01-18 11:37:26.481] [info] Got new work.
[2017-01-18 11:37:26.481] [info] GPU0 Got work, start mining.
[2017-01-18 11:37:26.855] [info] Using experimental highly optimized kernel code for Tahiti devices.
[2017-01-18 11:37:26.855] [info] Autodetected '--intensity 5' for device 2.
[2017-01-18 11:37:26.867] [info] [GPU2] Device info: {"id": "0/2" "name": "Tahiti" "platform": "AMD" "driver": "1912.5 (VM)"}
[2017-01-18 11:37:27.693] [error] Found solution index out of range: 1933034935
[2017-01-18 11:37:31.615] [info] [GPU0]  162.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:31.942] [info] [GPU1]  172.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:32.731] [info] [GPU2]  170.0 I/s 2.0 S/s (5s)
[2017-01-18 11:37:33.952] [info] [Total]  499.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:36.519] [error] Found solution index out of range: 670200178
[2017-01-18 11:37:36.643] [info] [GPU0]  160.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:36.949] [info] [GPU1]  168.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:37.746] [info] [GPU2]  169.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:38.969] [info] [Total]  500.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:41.657] [info] [GPU0]  160.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:41.956] [info] [GPU1]  170.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:42.754] [info] [GPU2]  170.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:42.987] [error] Found solution index out of range: 513475636
[2017-01-18 11:37:42.987] [error] Found solution index out of range: 513475636
[2017-01-18 11:37:43.981] [info] [Total]  499.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:46.714] [info] [GPU0]  160.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:46.970] [info] [GPU1]  170.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:47.770] [info] [GPU2]  171.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:48.997] [info] [Total]  503.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:49.937] [error] Found solution index out of range: 3835538899
[2017-01-18 11:37:51.717] [info] [GPU0]  162.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:51.986] [info] [GPU1]  171.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:52.778] [info] [GPU2]  169.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:53.998] [info] [Total]  499.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:55.432] [error] Found solution index out of range: 1653272006
[2017-01-18 11:37:56.742] [info] [GPU0]  159.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:57.012] [info] [GPU1]  170.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:57.809] [info] [GPU2]  169.0 I/s 0.0 S/s (5s)
[2017-01-18 11:37:59.010] [info] [Total]  499.0 I/s 0.0 S/s (5s)
[2017-01-18 11:38:01.755] [info] [GPU0]  161.0 I/s 0.2 S/s (5s)
[2017-01-18 11:38:02.021] [info] [GPU1]  169.0 I/s 0.0 S/s (5s)
[2017-01-18 11:38:02.833] [info] [GPU2]  170.0 I/s 0.0 S/s (5s)
[2017-01-18 11:38:04.021] [info] [Total]  499.0 I/s 0.2 S/s (5s)
```

**Comment by bpvarsity** [comment]
Mine does that with 290s but after 3+ minutes its starts working fine.

**Comment by bpvarsity** [comment]
Anyone find a fix yet?

**Comment by bpvarsity** [comment]
Getting this on two of my ethos machines as well as windows.

**Comment by bpvarsity** [comment]
Looks like it may be fixed in new version :)

**Comment by Optiminer** [comment]
Should be fixed in 1.6.0.
